source/src/ZbW.DesignPatterns/Strategy/Sale.cs

```diff
@@ -18,5 +20,21 @@ public decimal GetTotal()
             var discount = _pricingStrategy.GetDiscount(this);
             return Amount - discount;
         }
+
+        public override Prototypes ShallowCopy()
+        {
+            return (Sale) this.MemberwiseClone();
+        }
+
+        public override Prototypes DeepCopy()
+        {
+            // is ShallowCopy() Implemented, then call this.ShallowCopy()
+            Sale _sale = (Sale)this.MemberwiseClone(); // --> I dont need this then
```

Comment:
Die einfachste jedoch nicht die performanteste Art ein DeepCopy zumachen in C# ist per `MemoryStream` und `BinaryFormatter` und serialisieren und deserialisieren.

```
 using (MemoryStream memoryStream = new MemoryStream())
 {
     var binaryFormatter = new BinaryFormatter();
     binaryFormatter.Serialize(memoryStream, objectToCopy);
     memoryStream.Seek(0, SeekOrigin.Begin);
     return (T) binaryFormatter.Deserialize(memoryStream);
}
```